SignalServiceKit/src/Storage/KeychainStorage.swift

```diff
@@ -94,6 +94,10 @@ public class SSKDefaultKeychainStorage: NSObject, SSKKeychainStorage {
         var error: NSError?
         let result = SAMKeychain.deletePassword(forService: service, account: key, error: &error)
         if let error = error {
+            // If deletion failed because the specified item could not be found in the keychain, consider it success.
+            if error.code == errSecItemNotFound {
```

Comment:
A guard statement would better capture this pattern.

Reply:
Thanks for the feedback. Are you suggesting replacing the new `if` statement with a `guard` statement like this?
```swift
guard error.code != errSecItemNotFound else {
    return
}
```

-----

There's also the possibility of condensing this whole section to
```swift
if let error = error {
    guard error.code == errSecItemNotFound else {
        throw KeychainStorageError.failure(description: "\(logTag) error removing data: \(error)")
    }
}
```
The problem with the second approach would be the fall-through to the `guard result` statement below, which would require further refactoring to avoid mistakenly throwing that error instead.